### DISCUSSION TITLE
Images hosted on Google Drive are not displaying on a website hosted on GitHub Pages

### DISCUSSION THREAD

**Discussion by Viviraje** [community/community#86986] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello GitHub Pages community,
I'm having an issue with the images (hosted on public Google Drive folder) on my GitHub Pages-hosted website. The images aren't loading in any browser, not on desktop or mobile.
Last two months was working fine, and I did no changes. But now January 2024 are not displaying.
I've checked the image paths and they're correct. I've also checked the file permissions and the image files have the correct permissions. I've cleared the browser cache, but the problem is still there.
If anyone can help me fix this problem I would greatly appreciate it.

Link to website: https://viviraje.com
Repository: https://github.com/Viviraje/viviraje.github.io

Thank you,

**Comment by faridnec** [comment]
Hey, Instead of using the link to the Google Drive folder, use direct links to the images. To get the direct link, right-click on the image file in Google Drive, select "Get shareable link," and then replace /view or /edit in the link with /uc?export=view.

```bash
Before: https://drive.google.com/file/d/your_file_id/view
After: https://drive.google.com/uc?export=view&id=your_file_id
```
or instead of linking directly to the images, try embedding them using HTML <img> tags with the src attribute pointing directly to the Google Drive image URL.
```html
<img src="https://drive.google.com/uc?export=view&id=your_file_id" alt="Description">
```
hope this helps

**Reply by Viviraje** [reply]
Thanks for your answer,
They look like you said. The were working until 2024, now the do not display:
If you could have a look to my index.html?
https://github.com/Viviraje/viviraje.github.io
Thank you!

**Reply by Swarup4545** [reply]
thank you so much @code-PA-32

**Comment by code-PA-32** [comment, marked as answer]
See the example of the change:

OLD LINE OF CODE: `https://drive.google.com/uc?export=view&id={ID}`
NEW LINE OF CODE: `https://drive.google.com/thumbnail?id={ID}&sz=w1000`

**Reply by Viviraje** [reply]
Many thanks! Now it works!
Why the old way does not work any more?
Where can I get this kind of news to do not get old fashioned on my html code?
Thank you very much!

**Reply by code-PA-32** [reply]
This happens since January 24 due to some Google updates 
[Check it here Google issue tracker](https://issuetracker.google.com/issues/319531488?pli=1)

**Reply by Viviraje** [reply]
Thanks for your reply! Very helpful.
Have a great day!
Best regards

**Reply by JatiMuar** [reply]
NEW LINE OF CODE: https://drive.google.com/thumbnail?id={ID}&sz=w1000
👆 Image loaded resolution very low. How to load image for full resolution ?

**Reply by ZMYaro** [reply]
The `/thumbnail` endpoint doesn't seem to work consistently if there are too many images on the page.  I tried to load a page that had 10+ images embedded with that new format, and it rate limited me.  I tried a bit later, and it worked, so it doesn't seem to consistently do that, but I can't ship this if there is a chance GDrive will do that.

**Reply by Mr-Sim** [reply]
It's working on  desktop devices but not on mobile devices. Why google can't just keep things simple...

**Comment by YasserAhmed1** [comment]
Here is the URL that worked for me:
`<img src="https://lh3.googleusercontent.com/d/[FILE ID]=s[xxx]?authuser=0" >`
Replace [FILE ID] with google drive file id and replace [xxx] with the image size such as 500 or 900 (the bigger number the higher resolution)

**Reply by MURGGAR** [reply]
thanks this was very help full

**Reply by Davidmendoza1987** [reply]
This worked for me as of 2024-08-22

**Reply by SujitVidyutva** [reply]
> <img src="https://lh3.googleusercontent.com/d/[FILE ID]=s[xxx]?authuser=0" >

i want to used this but i face with this "(Too Many Requests)", can you please tell me i am using Remix framework, when i am using uc?export=view&id showing 403 forbidden. Can you please help me out!

**Reply by Viviraje** [reply]
For me it is working perfectly. You can check my website working with those lines, loading the pictures from google drive: viviraje.com
